CodeEditModules/Modules/AppPreferences/src/Sections/ThemePreferences/Model/ThemeModel.swift

```diff
@@ -143,10 +166,33 @@ public final class ThemeModel: ObservableObject {
 
                 // if there already is a selected theme in `preferences.json` select this theme
                 // otherwise take the first in the list
-                self.selectedTheme = self.themes.first { $0.name == prefs.theme.selectedTheme } ?? self.themes.first
+                self.selectedDarkTheme = self.darkThemes.first { $0.name == prefs.theme.selectedDarkTheme } ?? self.darkThemes.first
+                self.selectedLightTheme = self.lightThemes.first { $0.name == prefs.theme.selectedLightTheme } ?? self.lightThemes.first
+                
+                // For selecting the default theme, doing it correctly on startup requires some more logic
+                let userSelectedTheme = self.themes.first { $0.name == prefs.theme.selectedTheme }
+                if userSelectedTheme != nil {
+                    self.selectedTheme = userSelectedTheme
+                } else {
+                    print(colorScheme == .dark)
+                    if colorScheme == .dark {
+                        self.selectedTheme = self.selectedDarkTheme
+                    } else {
+                        self.selectedTheme = self.selectedLightTheme
+                    }
+                }
```

Comment:
So, this part is not working the way I want. It seems that ColorScheme is not yet initialized by the application at this point. The `colorScheme` variable is always `.light` even if the system appearance is dark. What happens, is that the app will always open with a white theme at the first start, instead of using the appearance of the system.

The question is where should this code be executed, so that it works haha

Reply:
I ran into the same thing. I ended up calling the function in AppDelegate when the app first initializes. We probably need some kind of initialization file we can break these things in so AppDelegate doesn't get so big. If anyone has any ideas or thoughts around this I am open to them as well.

Reply:
I initially implemented the logic this way too as it made more sense, but I ran into the same issue. I found that colorScheme only worked at detecting the right theme when the variable was declared and tested inside of a View file, and not a general Swift file (like this ThemeModel).

Reply:
@adsalad That was my experience as well. 
@matthijseikelenboom You may mention this in #help on Discord.

Reply:
@austincondiff I've found a fix. Not 100% sure if it's the most elegant way, but it works :)